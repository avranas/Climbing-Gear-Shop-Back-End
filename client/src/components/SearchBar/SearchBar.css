
#search-bar {
  position: relative;
  display: flex;
  height: 1.75rem;
  border-radius: 4px;
  border: solid;
  border-color: #BBBBBB;
  border-width: 2px;
  height: 3rem;
  width: 30rem;
  background-color: white;
}

#search-text:focus {
  outline: none;
}

#search-bar:focus-within {
  border: solid;
  border-width: 2px;
  border-color: blue;
}

#search-text {
  border: none;
  line-height: 5rem;
  width: 30rem;
  padding: 0.5rem;
}

#clear-button { 
  position: absolute;
  right: 3rem;
  top: 1rem;
  background-color: rgba(0,0,0,0);
  background-image: url("../../images/gray-x.png");
  background-size: cover;
  width: 1rem;
  height: 1rem;
  border: none;
}

#search-button {
  background-color: rgba(0,0,0,0);
  background-image: url("../../images/search.png");
  background-size: cover;
  width: 2rem;
  height: 2rem;
  margin: 0.4rem;
  border: none;
}
