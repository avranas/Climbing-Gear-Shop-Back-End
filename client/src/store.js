import { configureStore } from "@reduxjs/toolkit";
import userReducer from './slices/userSlice';
import productListSlice from './slices/productListSlice';
import notificationSlice from './slices/notificationSlice';
import productSlice from "./slices/productSlice";
import cartSlice from "./slices/cartSlice";
import newestCartItemSlice from "./slices/newestCartItemSlice";

export default configureStore({
  reducer: {
    user: userReducer,
    productList: productListSlice,
    product: productSlice,
    notifications: notificationSlice,
    cart: cartSlice,
    newestCartItem: newestCartItemSlice
  },
});