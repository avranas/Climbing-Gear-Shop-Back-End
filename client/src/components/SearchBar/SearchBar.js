import { useState } from "react";
import { useNavigate } from "react-router-dom";
import "./SearchBar.css";

const SearchBar = (props) => {
  const navigate = useNavigate();
  const [searchBarText, setSearchBarText] = useState("");

  const handleChange = (e) => {
    setSearchBarText(e.target.value);
  };

  const handleSubmit = async () => {
    if (searchBarText === "") {
      alert("Search bar can not be empty");
    } else {
      navigate(`/products/0?search=${searchBarText.toLowerCase()}`);
      setSearchBarText("");
    }
  };

  const handleKeyPress = (e) => {
    if (e.key === "Enter") {
      handleSubmit();
    }
  };

  const clearText = () => {
    setSearchBarText("");
  };

  return (
    <div id="search-bar" data-testid="searchbar">
      <input
        type="text"
        id="search-text"
        name="search"
        value={searchBarText}
        onChange={handleChange}
        onKeyUp={handleKeyPress}
      ></input>
      <button
        id="search-button"
        data-testid="searchbar-button"
        onClick={handleSubmit}
      />
      {searchBarText !== "" && <button id="clear-button" onClick={clearText} />}
    </div>
  );
};

export default SearchBar;
