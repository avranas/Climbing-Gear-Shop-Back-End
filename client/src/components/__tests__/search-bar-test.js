import { render, cleanup } from "@testing-library/react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import store from "../../store";
import SearchBar from "../SearchBar/SearchBar";

afterEach(() => {
  cleanup();
});

const searchBar = (
  <Provider store={store}>
    <BrowserRouter>
      <SearchBar />
    </BrowserRouter>
  </Provider>
);

test("Searchbar matches snapshot", async () => {
  render(searchBar)
  const tree = renderer.create(searchBar).toJSON();
  expect(tree).toMatchSnapshot();
});