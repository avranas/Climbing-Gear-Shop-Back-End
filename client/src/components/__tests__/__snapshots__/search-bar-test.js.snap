// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar matches snapshot 1`] = `
<div
  data-testid="searchbar"
  id="search-bar"
>
  <input
    id="search-text"
    name="search"
    onChange={[Function]}
    onKeyUp={[Function]}
    type="text"
    value=""
  />
  <button
    data-testid="searchbar-button"
    id="search-button"
    onClick={[Function]}
  />
</div>
`;
